src/main/java/org/jabref/logic/ai/models/EmbeddingModel.java

```diff
@@ -0,0 +1,104 @@
+package org.jabref.logic.ai.models;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.SimpleObjectProperty;
+
+import org.jabref.logic.l10n.Localization;
+import org.jabref.preferences.AiPreferences;
+
+import ai.djl.MalformedModelException;
+import ai.djl.huggingface.translator.TextEmbeddingTranslatorFactory;
+import ai.djl.repository.zoo.Criteria;
+import ai.djl.repository.zoo.ModelNotFoundException;
+import ai.djl.training.util.ProgressBar;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.output.Response;
+
+/**
+ * Wrapper around langchain4j embedding model.
```

Comment:
Can you use `{@link ...}`  to provide "hyperlinks" here? In IntelliJ, one can directly click on the linked element to be send to there.